apps/management-app/src/components/Authenticated/CDO/PanelTypesDetail/index.tsx

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { yupResolver } from '@hookform/resolvers/yup';
+import ArrowBackIcon from '@mui/icons-material/ArrowBack';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import { useCallback, useEffect, useMemo } from 'react';
+import { useForm } from 'react-hook-form';
+import { useNavigate, useParams } from 'react-router-dom';
+import * as yup from 'yup';
+import ControlledTextField from '@/components/Common/ControlledTextField';
+import {
+  useDeletePanelTypesMutation,
+  useLazyGetPanelTypeByIdQuery,
+  useUpdatePanelTypeMutation,
+} from '@/store/api/generalManagementApiSlice';
+import { isApiErrorResponse } from '@/store/api/helper';
+import { showError, showSuccess } from '@/utils/toast';
+import FormInputSkeleton from '../FormInputSkeleton';
+import StaticActionBar from '../StaticActionBar';
+
+interface FormData {
+  name: string;
+}
+
+const PanelTypesDetail = () => {
+  const { id } = useParams();
+  const navigate = useNavigate();
+
+  const [getPanelType, { isError }] = useLazyGetPanelTypeByIdQuery();
+
+  useEffect(() => {
+    if (isError) navigate(-1);
+  }, [isError, navigate]);
+
+  const schema = useMemo(
+    () =>
+      yup.object({
+        name: yup.string().required("Panel type's name is required"),
+      }),
+    [],
+  );
+
+  const {
+    control,
+    handleSubmit,
+    formState: { isDirty, isValid, isLoading },
+    reset,
+  } = useForm<FormData>({
+    resolver: yupResolver(schema),
+    mode: 'onChange',
+    defaultValues: async () => {
+      const panelType = await getPanelType(parseInt(id!)).unwrap();
+      return panelType ? { name: panelType.name } : { name: '' };
+    },
+  });
+
+  const [updatePanelType] = useUpdatePanelTypeMutation();
+
+  const handleUpdatePanelType = useCallback(
+    handleSubmit(async (data: FormData) => {
+      try {
+        await updatePanelType({ id: parseInt(id!), data }).unwrap();
```

Comment:
id! -> what does it mean?

Reply:
Non-null assertion, the page would never be reached if the id params was empty, typescript doesn't know this.